axum/src/routing/mod.rs

```diff
@@ -44,25 +42,32 @@ pub use self::method_routing::{
     trace_service, MethodRouter,
 };
 
+macro_rules! panic_on_err {
+    ($expr:expr) => {
+        match $expr {
+            Ok(x) => x,
+            Err(err) => panic!("{err}"),
+        }
+    };
+}
```

Comment:
This is just `.unwrap()`, no?

Reply:
Almos:

```
# unwrap
❯ cargo run
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: E', src/main.rs:5:7
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

# unwrap
❯ cargo run
thread 'main' panicked at 'something went wrong', src/main.rs:7:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I prefer the latter.

`unwrap` also prints the debug representation.